src/ModeTypeDetector.php

```diff
@@ -2,65 +2,28 @@
 
 namespace React\Filesystem;
 
-use Exception;
-use React\Filesystem\FilesystemInterface;
-use React\Filesystem\TypeDetectorInterface;
+use React\EventLoop\ExtUvLoop;
+use React\EventLoop\LoopInterface;
+use React\Filesystem\Node\DirectoryInterface;
+use React\Filesystem\Node\FileInterface;
+use React\Filesystem\Node\Unknown;
 
-class ModeTypeDetector implements TypeDetectorInterface
+final class ModeTypeDetector
 {
-    /**
-     * @var array
-     */
-    protected $mapping = [
-        0xa000 => 'constructLink',
-        0x4000 => 'dir',
-        0x8000 => 'file',
-    ];
+    private const FILE = 0x8000;
+    private const DIRECTORY = 0x4000;
+    private const LINK = 0xa000;
 
-    /**
-     * @var FilesystemInterface
-     */
-    protected $filesystem;
-
-    /**
-     * @param FilesystemInterface $filesystem
-     */
-    public function __construct(FilesystemInterface $filesystem)
+    public static function detect(int $mode): string
     {
-        $this->filesystem = $filesystem;
-    }
-
-    /**
-     * @param array $node
-     * @return \React\Promise\PromiseInterface
-     */
-    public function detect(array $node)
-    {
-        return $this->filesystem->getAdapter()->stat($node['path'])->then(function ($stat) {
-            return $this->walkMapping($stat);
-        });
-    }
-
-    protected function walkMapping($stat)
-    {
-        $promiseChain = \React\Promise\reject(new Exception('Unknown type'));
-        foreach ($this->mapping as $mappingMode => $method) {
-            $promiseChain = $promiseChain->otherwise(function () use ($stat, $mappingMode, $method) {
-                return $this->matchMapping($stat['mode'], $mappingMode, $method);
-            });
+        if (($mode & self::FILE) == self::FILE) {
```

Comment:
Since `$mode` is decimal and `FILE` and `DIRECTORY` constants are hexademical, they all should be converted to the same number system, in order to correctly operate and compare their values.

Possible solution:
```if ((decoct($mode) & sprintf("%o", self::FILE)) == sprintf("%o", self::FILE)) {```

Reply:
I've rechecked this operations again and I've noticed that your original solution works fine (values from different number systems are correctly compared, so it means that auto-convertion to one number system works out of the box).

Don't know what was wrong with my test enviroment earlier :/ Tests pointed me to problem with decimal/hexademical comparsion. And I fixed processing with this quick fix.

Sorry for confusion.

Reply:
No problem, thanks for looking into this :+1: